sds_data_manager/lambda_code/SDSCode/batch_starter.py

```diff
@@ -437,24 +504,47 @@ def lambda_handler(event: dict, context):
 
         # Start Batch Job execution for those that has all dependencies
         for downstream_data in downstream_instruments_to_process:
-            command = downstream_data["command"]
-            logger.info(f"Submitting job with this command - {command}")
+            # FYI, these are the keys the upstream_dependencies should contain:
+            # {
+            #     'instrument': 'swe',
+            #     'data_level': 'l0',
+            #     'descriptor': 'lveng-hk',
+            #     'start_date': '20231212',
+            #     'version': 'v001',
+            # },
+            batch_command = [
+                "--instrument",
+                downstream_data["instrument"],
+                "--data-level",
+                downstream_data["data_level"],
+                "--start-date",
+                downstream_data["start_date"],
+                "--version",
+                downstream_data["version"],
+                "--dependency",
+                f"{downstream_data['upstream_dependencies']}",
+                "--upload-to-sdc",
+            ]
+            logger.info(f"Submitting job with this command - {batch_command}")
             # NOTE: The batch job name should contain only
             # alphanumeric characters and hyphens.
-            level_to_process = command[3]
+            level_to_process = downstream_data["data_level"]
             # Eg. "codice-l1a-job"
-            job_name = f"{instrument}-{level_to_process}-job"
+            job_name = (
+                f"{downstream_data['instrument']}-{level_to_process}-"
+                f"{downstream_data['descriptor']}-job"
+            )
```

Comment:
I'm just thinking of this now but could we update the status table based on the job name? Parse the information needed out of this string. Add in the `start_date` and `version` and I think we'd be there.